Sources/PayPal/Enums/PayPalError.swift

```diff
@@ -0,0 +1,35 @@
+@_implementationOnly import PayPalCheckout
+
+#if canImport(PaymentsCore)
+import PaymentsCore
+#endif
+
+enum PayPalError {
+
+    static let domain = "PayPalErrorDomain"
+
+    enum Code: Int {
+        /// 0. An unknown error occurred.
+        case unknown
+
+        /// 1. Error returned from the PayPal Checkout SDK
+        case nativeCheckoutSDKError
+
+        /// 2. The buyer canceled the PayPal Checkout flow
+        case cancelled
```

Comment:
I don't think I agree with this being an error scenario. Users cancel the checkout flow all the time for any number of reasons - them doing so shouldn't be regarded as an error

Reply:
Merchants might very well record errors generated by our SDK to their own analytics platform, or make a UI change in response to a payment error that they would want to suppress if it's a checkout cancellation

Reply:
I keep going back and fourth on this, to be honest. I can see an argument for it being it's own case since it's not a success or error but also can see an argument for it falling into the error bucket as it's an unsuccessful completion of checkout. I think BT handles cancellations as an error today whereas Stripe has a cancelled case as we did previously (though theirs is called "user cancelled").

Reply:
Yep that is right - BT currently returns an error. I mean on one hand, merchants have been using that integration pattern for years and it hasn't been an issue. On the other hand, doesn't mean it couldn't better, especially now that we have Swift enums.

This is the _simpler_ way, since it doesn't require a custom Result type. Hmmmm .... I see both sides as well.

Some food for thought - what would Result types look like for other payment methods? I would imagine similar here: success, cancel, result.

Reply:
I think the result type would look different for other payment methods, since the idea of a user initiated cancellation doesn't really exist for flows that do not involve user interaction. For an API wrapping function, we will always have a successful response or a failure response - so having `success/failure` makes sense in taht context, but it feels like it would be sending the wrong message to the merchant to say that a user cancellation is an error. 

fwiw the JS SDK also treats cancellations separately from errors.

Reply:
It looks like [Android also treats cancellations separate from error or success](https://github.com/paypal/Android-SDK/blob/d2768255e848507cba91892e1a3c867f65c2bcf9/PayPalCheckout/src/main/java/com/paypal/android/checkout/PayPalCheckoutResult.kt). Anyone opposed to us going back to that?

Reply:
Nope, that sounds good to me!